### PR TITLE
1003 ref list used for 2022 only

### PR DESCRIPTION
**Pull request by AnneONS**
## Pull Request submission

After discussions with Leela, Dan and Ana, and analysis [RDRP-991](https://jira.ons.gov.uk/browse/RDRP-991) we have determined that after much to and fro, the reference list isn't needed for 2023. As initially thought during the discovery phase, it was only needed as a fix for 2022 data.

In this PR

1) removed the 817 mapper from mapper folders (created a v2 for this)

2) removed references to the 817 mapper in the config for 2023

3) update the pipeline so it only looks for, loads and joins the ref list when running survey_year = 2022
4) As the itl1_detailed mapper was being read in but not used in the pipeline, I've also removed this.

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/mapping/mapping_main.py [hunk 1]
```diff
@@ -84,9 +84,9 @@ def run_mapping(
     )
     val.validate_many_to_one(sic_pg_num, "SIC 2007_CODE", "2016 > Form PG")
 
-    # For survey year 2022 and 2023 it's necessary to update the reference list
-    # TODO: include in the config the ability to decide for future years about ref list
-    if config["years"]["survey_year"] in [2022, 2023]:
+    # For survey year only 2022 ait's necessary to update the reference list
```
Comment (zorge69) [inline src/mapping/mapping_main.py]: typo in a comment, corrected

File: src/staging/staging_main.py [hunk 2]
```diff
@@ -209,15 +209,6 @@ def run_staging(  # noqa: C901
 
     StagingMainLogger.info("Backdata File Loaded Successfully...")
 
-    # Loading ITL1 detailed mapper
```
Comment (zorge69) [inline src/staging/staging_main.py:212]: Please check that itl1_detailed_mapper is removed from the directory in DAP and from the res_dev_runner config.

File: tests/test_mapping/test_mapping_helpers.py [hunk 3]
```diff
@@ -4,7 +4,6 @@
 
 # Local Imports
 from src.mapping.mapping_helpers import (
-    mapper_null_checks,
```
Comment (zorge69) [inline tests/test_mapping/test_mapping_helpers.py:7]: Remove the function mapper_null_checks from src\mapping\mapping_helpers.py

File: tests/test_mapping/test_mapping_helpers.py [hunk 4]
```diff
@@ -267,11 +266,16 @@ class TestValidateMappingFilenames(object):
     def test_validate_mapper_config_raises_file_incorrect(self):
         config = {
             'years' : {'survey_year': 2022,},
-            '2022_mappers': {'itl_mapper_path': "itl_2022.cs",
-            'ultfoc_mapper_path': "BERD_2022_ultfoc.csv",},
-            '2023_mappers': {'itl_mapper_path': "itl_2023.csv",
-            'ultfoc_mapper_path': None,}
-            } 
+            '2022_mappers': {
```
Comment (zorge69) [inline tests/test_mapping/test_mapping_helpers.py:269]: We need only one section for mappers in the user config, <this year>_mappers. I am happy to pass it for now but we need to decide what to do,

**Review by zorge69 — CHANGES_REQUESTED** [review]

File: src/staging/staging_main.py (hunk 2 above)
Comment (AnneONS) [reply src/staging/staging_main.py:212]: I'll do it in DAP when we do the new release.

File: tests/test_mapping/test_mapping_helpers.py (hunk 4 above)
Comment (AnneONS) [reply tests/test_mapping/test_mapping_helpers.py:269]: you are right, we'll sort it out

**Review by zorge69 — APPROVED** [review]

**Pull request merged**